webapp/src/shared/components/Avatar/EditableAvatar.tsx

```diff
@@ -0,0 +1,48 @@
+import './EditableAvatar.css';
+import { AvatarProps } from './Avatar';
+import useDrag from '../../hooks/UseDrag';
+import { Text, TextColor, TextVariant, TextWeight } from '../index';
+
+export function EditableAvatar({ url }: AvatarProps) {
```

Comment:
What do we think about cropping the avatar before upload? This way, we could use our current Avatar components and the API endpoints as they are.

I am concerned that with this approach, we save frontend logic in the backend. We have to persist the position of how the user 'crop' their avatar in the database, and when displaying it, we have to use that position again.

Reply:
We query the BE too get all user data. X and y of the Avatar would be just two more number fields: doesn't seem like an overkill. We only need to modify the patch negros for users, and add two fields . And like this, it's reversible, if the user devices he wants a different cropping.

Reply:
About keeping the Avatar the way it currently is: I don't know about yours, but mine is completely deformed. The way it is only works if your picture has some particular proportions.

Reply:
Also: we don't really save logic
 We save data that we intend to persist, as we should. It just happens to be about the display of things.

Reply:
I don't understand why we want to make the cropping reversible. If the user wants different cropping, they could upload the image again.

Anyway, even if I think the current solution is not optimal, it's pretty creative and could work.